src/wallet/spend.rs

```diff
@@ -20,7 +20,12 @@ pub enum Destination {
     /// Sweep
     Sweep(Address),
     /// Multi
-    Multi(Vec<(Address, Amount)>),
+    Multi {
+        /// List of outputs (address, amounts)
+        outputs: Vec<(Address, Amount)>,
+        /// OP_RETURN data, used to create a OP_RETURN TxOut
+        op_return_data: Option<Box<[u8]>>,
+    },
```

Comment:
Would love to explore some way where we can do this without changing the spend API. 

We can create another variant for fidelity, which will take one single output and op return data. That way, all other spend API call sites will remain the same.